application-collabora-ui/src/main/resources/Collabora/Code/Main.xml

```diff
@@ -197,6 +224,26 @@
     <property>
       <code>require(['jquery'], function($) {
   const collaboraPath ='/rest/collabora/files/';
+  const iframeWindow = document.getElementById('collaboraViewer').contentWindow;
+  var isDocModified = false;
+
+  var receiveMessage = function(event) {
+    // Skip messages that are not from this iframe.
+    if (event.source !== iframeWindow) {
+      return;
+    }
+    var msg = JSON.parse(event.data);
+    if (!msg) {
+      return;
+    }
+
+    if (msg.MessageId == 'App_LoadingStatus' &amp;&amp; msg.Values &amp;&amp; msg.Values.Status == 'Document_Loaded') {
+      // Notify the server that postMessage is going to be used.
+      iframeWindow.postMessage(JSON.stringify({ 'MessageId': 'Host_PostmessageReady' }), '*');
```

Comment:
Don't we know the Collabora editor origin? The documentation https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#security_concerns recommends to always specify an exact target origin, instead of '*'.